java/custom/com/sun/midp/security/Permissions.java

```diff
@@ -0,0 +1,294 @@
+     *
+     * @exception SecurityException if the permission is invalid
+     */
+    public static String getName(int permission) {
+        // The code that uses this method doesn't actually use the return value, but
+        // passes it to Permissions.getId. So we can return anything.
+        return "com.sun.midp";
+    }
+
+    /**
+     * Get the dialog title for a permission.
+     *
+     * @param permission permission number
+     *
+     * @return Resource constant for the permission dialog title
+     * @exception SecurityException if the permission is invalid
+     */
+    public static String getTitle(int permission) {
+        return "Title";
+    }
+
+    /**
+     * Get the question for a permission.
+     *
+     * @param permission permission number
+     *
+     * @return Resource constant for the permission question
+     *
+     * @exception SecurityException if the permission is invalid
+     */
+    public static String getQuestion(int permission) {
+        return "Question";
+    }
+
+    /**
+     * Get the oneshot question for a permission.
+     *
+     * @param permission permission number
+     *
+     * @return Resource constant for the permission question
+     *
+     * @exception SecurityException if the permission is invalid
+     */
+    public static String getOneshotQuestion(int permission) {
+        return "Oneshot Question";
+    }
+
+    /**
+     * Get the ID of a permission.
+     *
+     * @param name permission name
+     *
+     * @return permission ID
+     *
+     * @exception SecurityException if the permission is invalid
+     */
+    public static int getId(String name) {
+        // Returns the ID of the permission. The callers will use this ID to check the
+        // permission in the permissions array returned by Permissions::forDomain.
+        return 0;
+    }
+
+    /**
+     * Determine if a domain is a trusted domain.
+     *
+     * @param domain Binding name of a domain
+     *
+     * @return true if a domain is trusted, false if not
+     */
+    public static boolean isTrusted(String domain) {
+        // Always return true to make Java think the MIDlet domain is trusted.
+        // We rely on the web security model, so we don't need to ask permissions
+        // to the user.
+        return true;
+    }
+
+    /**
+     * Returns domain for unsigned midlets.
+     *
+     * @return domain name
+     */
+    public static String getUnsignedDomain() {
+        return UNIDENTIFIED_DOMAIN_BINDING;
+    }
+
+    /**
+     * Create a list of permission groups a domain is permitted to perform.
+     *
+     * @param name binding name of domain
+     *
+     * @return 2 arrays, the first containing the maximum level for each
+     *     permission, the second containing the default or starting level
+     *     for each permission supported
+     */
+    public static byte[][] forDomain(String name) {
+        NUMBER_OF_PERMISSIONS = PermissionsStrings.PERMISSION_STRINGS.length + 2;
```

Comment:
It'd better to keep all the original comments about how to calculate the permission number:
// The 2 is the two hardcoded MIPS and AMS permissions.